### ISSUE TITLE
Support WebVTT files

### ISSUE THREAD

**Issue by marnusw** [gsantiago/subtitle.js#20]
I need to process either SRT or VTT files depending on what is available. If I work on adding support for VTT files (which is superset of SRT) would that be something you would consider merging?

I think just stripping the web VTT header and styling when parsing will be a good start. Then VTT input will be supported with SRT output. A further step might be to keep the additional info in the captions array as well, from where it could then be added back in `stringify`.

I'll open a PR once I've made some progress over the coming days.

**Comment by gsantiago** [comment]
PR's are welcome. Thank you!

**Comment by marnusw** [comment]
Continued in #21

**Comment by marnusw** [comment]
With WebVTT supported as input the next step would be to support manipulation and stringify of VTT as well.

The simplest case could be handled by passing another parameter to `stringify` to indicate whether SRT or WebVTT output should be used. Then all that needs to be done is add the `WEBVTT` header and use a `.` for the ms separator instead of `,`. If the caption objects contain a `settings` prop those could also be added.

There are a number of additional features supported by WebVTT though: text headers, text in indexes, comments, and embedded styles. I doubt these are critical features in most use cases, but supporting them would make this tool feature complete. To keep track of it all between `parse` and `stringify` an object more complex than an array of cues will be necessary, though, which would change the API.

Something like this could work:

```javascript
{
  header: '- Text next to WEBVTT',
  styles: [/* STYLE strings */],
  comments: [/* NOTE strings */],
  cues: [{
    start: 20000, // time in ms
    end: 24400,
    text: 'Bla Bla Bla Bla',
    settings: '...'
  }],
  // only problem is the order of cues and comments
  // will have to be maintained, so they have to be in
  // the same array; maybe with a separate array 
  // containing only cues.
}
```

Actually, perhaps that could be kept in a single array if each object has a `type` property; no type could default to being a cue which would make it backward compatible. Types could be `HEADER`, `STYLE`, `NOTE`, and `CUE`.

I'm just thinking out loud and still even debating with myself whether it's worth the effort to support all this though. As long as more complex input parses properly the stringify operation doesn't necessarily need all this.

I'll work on the simple case of exporting VTT format next if you like the idea of adding an additional parameter to `stringify` though.

**Comment by gsantiago** [comment]
I prefer a new function called something like `stringifyVTT` instead of pass another parameter to `stringify`. What do you think?

**Comment by marnusw** [comment]
That actually does make sense, and if the need arises to have a single function that can output different formats it can be a higher-order function that uses one of the other. (Makes me wonder if parsing VTT should have been done similarly, but at least it didn't require too many modifications to the existing functions.)

Thanks for merging #21. I should mention I believe it will still fail if the VTT file contains `NOTE` or `STYLE` sections. I noticed you're making some changes to the parse function. Once that has been done we might look at skipping over those sections to make it work for all cases, or to extract those as mentioned above.

**Comment by gsantiago** [comment]
Actually, I'm not working on the `parse` anymore. You can ignore my commits at development branch. I will focus now on solve the #18 and #19 issues.

So, there are no problems if you want to improve the `parse` or any other function for now.

**Comment by marnusw** [comment]
I think this issue can be closed by #24 for now.

**Comment by gsantiago** [comment]
@marnusw thanks!
